pulp_container/app/replica.py

```diff
@@ -0,0 +1,56 @@
+from pulpcore.plugin.replica import Replicator
+from pulpcore.plugin.util import get_url
+
+from pulp_glue.container.context import (
+    PulpContainerDistributionContext,
+    PulpContainerRepositoryContext,
+)
+
+from pulp_container.app.models import ContainerDistribution, ContainerRemote, ContainerRepository
+from pulp_container.app.tasks import synchronize as container_synchronize
+
+
+class ContainerReplicator(Replicator):
+    distribution_ctx_cls = PulpContainerDistributionContext
+    repository_ctx_cls = PulpContainerRepositoryContext
+    publication_ctx_cls = None
+    remote_model_cls = ContainerRemote
+    repository_model_cls = ContainerRepository
+    distribution_model_cls = ContainerDistribution
+    distribution_serializer_name = "ContainerDistributionSerializer"
+    repository_serializer_name = "ContainerRepositorySerializer"
+    remote_serializer_name = "ContainerRemoteSerializer"
+    app_label = "container"
+    sync_task = container_synchronize
+
+    def sync_params(self, repository, remote):
+        """Returns a dictionary where key is a parameter for the sync task."""
+        return dict(
+            remote_pk=str(remote.pk),
+            repository_pk=str(repository.pk),
+            signed_only=False,  # TODO not sure
+            mirror=True,
+        )
+
+    def url(self, upstream_distribution):
+        if upstream_distribution["secure"]:
```

Comment:
can't you extract this information from UpstreamPulp.base_url?

Reply:
This is a great point! Then, we can simply close the issue https://github.com/pulp/pulp_container/issues/1213 as "not needed".

Reply:
Do we need to expose `base_url` on the serializer?

Reply:
@ipanova @lubosmj we cannot do that, as the upstream_distribution parameter is a dictionary

maybe exposing base_url on the serializer would do the trick, should we do that?

Reply:
That is what I was asking for. Try removing the set difference.

Reply:
it's being passed into the the replica context https://github.com/pulp/pulpcore/blob/main/pulpcore/app/tasks/replica.py#L41

Reply:
How can I access it inside container's replica.py tho?